modules/rest/src/main/java/org/apache/ignite/rest/presentation/json/JsonConverter.java

```diff
@@ -35,6 +52,213 @@
         return gson.toJson(obj);
     }
 
+    /** */
+    public static ConfigurationVisitor<JsonElement> jsonVisitor() {
+        return new ConfigurationVisitor<JsonElement>() {
+            /** */
+            private final Deque<JsonObject> jsonObjectsStack = new ArrayDeque<>();
+
+            /** {@inheritDoc} */
+            @Override public JsonElement visitLeafNode(String key, Serializable val) {
+                JsonElement jsonLeaf = toJsonLeaf(val);
+
+                if (!jsonObjectsStack.isEmpty())
```

Comment:
I would suggest extracting this `if` into a function, this pattern is repeated quite often

Reply:
Done